Assets/Mapzen/Unity/Editor/RegionMapEditor.cs

```diff
@@ -53,14 +53,25 @@ public override void OnInspectorGUI()
                 {
                     LogWarnings();
 
-                    map.DownloadTiles();
+                    map.DownloadTilesAsync();
                 }
                 else
                 {
                     LogErrors();
                 }
             }
 
+            if (map.HasPendingTasks())
+            {
+                // Go through another OnInspectorGUI cycle
+                Repaint();
```

Comment:
So if I understand this part, it's causing continuous `Repaint()`s in a loop while the tile data is building on a background thread? If so, that seems like a lot of CPU time spent doing unneeded rendering. I suppose it does technically keep the UI interactive though. Is there no way to trigger an event on the main thread when the background threads complete their work?

Reply:
I'm starting to understand why this is tricky - the background threads can't safely call into the Unity framework to trigger the subsequent tasks :\

I did get this working with a Coroutine started from `RegionMap`. The problem with this approach is that in the editor, Couroutines don't seem to update regularly like they do in a player - which makes sense given that the editor doesn't run in a continuous loop.

I'm ok with merging as-is now, seeing as it's not easy to work around this.

Reply:
Yes, I'll give a look again into an alternative today and let you know if I find anything. I also tried the Coroutine way and figured the same.